### PR TITLE
add automated tests for presearch and algorand

### PR DESCRIPTION
**Pull request by A-Harby**
### Description

Add the automated tests for algorand and presearch application, (following same code struture as https://github.com/threefoldtech/tfgrid-sdk-ts/pull/2890).

The tests are different in that they ssh to the deployed VM and assert that the application is running with the correct version and network.

### CODE REVIEW

**Review by Mahmoud-Emad — APPROVED** [review]

**Review by amiraabouhadid — APPROVED** [review]

**Review by zaelgohary — CHANGES_REQUESTED** [review]
I tried running Algorand test and it failed w this err. Is this normal? Same with Presearch.

image

**Comment by A-Harby** [comment]
> I tried running Algorand test and it failed w this err. Is this normal? Same with Presearch.
> 
> image

Did you use the correct SSH key type? you should use the ssh-ed25519 not the normal one.

**Review by zaelgohary — COMMENTED** [review]
The test would fail if SSH key needs permission.

image

**Review by 0oM4R — COMMENTED** [review]
image

this [package](https://www.npmjs.com/package/semver#prerelease-identifiers) may help

**Review by 0oM4R — COMMENTED** [review]
and please notice that the current logic overwrite the provided config if one of the `mnemonic`, `network`, or `ssh_key` are not provided in env.

we may change that logic to get the missing `only` form config.json 

https://github.com/threefoldtech/tfgrid-sdk-ts/blob/828f0a0efacfd8a27e58a5befcd931782d2885f7/packages/grid_client/tests/client_loader.ts#L15-L26

File: packages/grid_client/tests/config.json
Comment (0oM4R) [inline packages/grid_client/tests/config.json:1]: remove those config please

File: packages/grid_client/tests/client_loader.ts [hunk 1]
```diff
@@ -11,17 +11,18 @@ const mnemonic = env.MNEMONIC;
 const storeSecret = env.STORE_SECRET;
 const ssh_key = fs.readFileSync(os.homedir() + "/.ssh/id_ed25519.pub", "utf-8");
 let config;
+const defaultConfig = JSON.parse(fs.readFileSync(path.join(__dirname, "./config.json"), "utf-8"));
 
 if (!network || !mnemonic || !ssh_key) {
```
Comment (0oM4R) [inline packages/grid_client/tests/client_loader.ts:16]: this won't fix the issue, 
Screenshot from 2024-07-04 08-10-28

i think its better to revert this changes and mention on docs that all credential must be in one place

**Review by 0oM4R — COMMENTED** [review]
image

presearch failed

**Comment by 0oM4R** [comment]
can you please validate if the machine that runs these tests has ygg installed and configured, 
I forgot to install ygg and got some connection errors, It was unclear if it came from the deployment or from my side

**Comment by A-Harby** [comment]
> can you please validate if the machine that runs these tests has ygg installed and configured, I forgot to install ygg and got some connection errors, It was unclear if it came from the deployment or from my side

As we discussed, I'm going to mention that on the Readme to make sure that ygg is needed to be installed with all of the other needed packages.

Also, I added a few tries in the trim of the Docker hash: https://github.com/threefoldtech/tfgrid-sdk-ts/pull/3030/commits/a3b4659951b9855546386019f4b30059e6116f8b.

And I have added the required notes here: https://github.com/threefoldtech/tfgrid-sdk-ts/pull/2890/commits/e8e35cba245a7c5f34c41ced7ce9cd733a689767.

**Review by 0oM4R — APPROVED** [review]

**Review by zaelgohary — APPROVED** [review]
Good job! @A-Harby

**Pull request merged**